### DISCUSSION TITLE
Cannot access ghcr.io

### DISCUSSION THREAD

**Discussion by simPod** [renovatebot/renovate#24982] [Request Help]
### How are you running Renovate?

Self-hosted Renovate

### If you're self-hosting Renovate, tell us what version of Renovate you run.

37.3.3

### If you're self-hosting Renovate, select which platform you are using.

gitlab.com

### Was this something which used to work for you, and then stopped?

It used to work, and then stopped

### Describe the problem

It used to work on v35.

Renovate cannot access ghcr.io, see logs for more info. It gives e403. I even tried using token with full permissions in `GITHUB_COM_TOKEN`.

![image](https://github.com/renovatebot/renovate/assets/327717/e1943b05-064a-46a4-a1b6-4a95891652c9)

### Relevant debug logs

<details><summary>Logs</summary>

```
DEBUG: GET https://ghcr.io/token?scope=repository%3Acloudflare%2Fpint%3Apull&service=ghcr.io = (code=ERR_NON_2XX_3XX_RESPONSE, statusCode=403 retryCount=0, duration=165) (repository=cdn/mon/app)
DEBUG: Not allowed to access docker registry (repository=my/repo)
       "registryHost": "https://ghcr.io",
       "dockerRepository": "cloudflare/pint"
DEBUG: Response code 403 (Forbidden) (repository=my/repo)
       "err": {
         "name": "HTTPError",
         "code": "ERR_NON_2XX_3XX_RESPONSE",
         "timings": {
           "start": 1696323229401,
           "socket": 1696323229401,
           "lookup": 1696323229401,
           "connect": 1696323229413,
           "secureConnect": 1696323229436,
           "upload": 1696323229444,
           "response": 1696323229566,
           "end": 1696323229566,
           "phases": {
             "wait": 0,
             "dns": 0,
             "tcp": 12,
             "tls": 23,
             "request": 8,
             "firstByte": 122,
             "download": 0,
             "total": 165
           }
         },
         "message": "Response code 403 (Forbidden)",
         "stack": "HTTPError: Response code 403 (Forbidden)\n    at Request.<anonymous> (/opt/containerbase/tools/renovate/37.3.3/node_modules/got/dist/source/as-promise/index.js:118:42)\n    at processTicksAndRejections (node:internal/process/task_queues:95:5)",
         "options": {
           "headers": {
             "user-agent": "RenovateBot/37.3.3 (https://github.com/renovatebot/renovate)",
             "accept": "application/json",
             "authorization": "***********",
             "accept-encoding": "gzip, deflate, br"
           },
           "url": "https://ghcr.io/token?scope=repository%3Acloudflare%2Fpint%3Apull&service=ghcr.io",
           "hostType": "docker",
           "username": "",
           "password": "",
           "method": "GET",
           "http2": false
         },
         "response": {
           "statusCode": 403,
           "statusMessage": "Forbidden",
           "body": {"errors": [{"code": "DENIED", "message": "denied"}]},
           "headers": {
             "content-type": "application/json",
             "docker-distribution-api-version": "registry/2.0",
             "date": "Tue, 03 Oct 2023 08:53:49 GMT",
             "content-length": "50",
             "x-github-request-id": "CEC4:6C68:81075B:84DBCE:651BD69D",
             "connection": "close"
           },
           "httpVersion": "1.1",
           "retryCount": 0
         }
       }
DEBUG: Failed to get authHeaders for getTags lookup (repository=my/repo)
```

</details>

### Have you created a minimal reproduction repository?

I have explained in the description why a minimal reproduction is impossible

**Comment by simPod** [comment]
I actually wonder that it might not have worked on v35. I guess there was no support for ghcr since I have not seen any update ever there.

**Comment by viceice** [comment]
is that a private image?

your renovate token needs packages read access.

**Reply by simPod** [reply]
No, it's public. https://github.com/cloudflare/pint/pkgs/container/pint

I have tried giving the token the full access as well. Currently I have read access for packages:

![image](https://github.com/renovatebot/renovate/assets/327717/92576d58-a018-4894-a690-6c5f0f2c5f49)

but the issue persists.

**Reply by rarkins** [reply]
Public GHCR.io images should even without a github token

**Reply by simPod** [reply]
Exactly I thought the same. So what might be the issue? When I try simple `curl` from the machine to the url from the logs it returns the output and status code 200. There is not much more that can be read from the logs 🤔

**Reply by viceice** [reply]
renovate is working fine for our own ghcr images, so it must be something with your renovate config or environment.

can you reproduce it with the hosted GitHub app?
